### PR TITLE
Bugfix/redirect url

### PR DESCRIPTION
**Pull request by Hydrock**
this PR is for https://github.com/dwyl/hapi-error/issues/69

-----------------------------------------------------------------------------

Hapi server used Node URL.

if we look at the old documentation ([v4](https://nodejs.org/docs/latest-v4.x/api/url.html)) we will see the **path** key

But in earlier versions of NodeJS ([v6](https://nodejs.org/docs/latest-v6.x/api/url.html)) - the key **path** is missing

I'll try something like this:
```
+ '?redirect=' + (request.url.pathname + request.url.search)
```

**search**: The 'query string' portion of the URL, including the leading question mark.
Example: '?query=string'
**pathname**: The path section of the URL, that comes after the host and before the query, including the initial slash if present. No decoding is performed.
Example: '/p/a/t/h'

These parameters are present in the old version of NodeJS

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -106,24 +110,25 @@ exports.plugin = {
             return reply.response(res.output.payload).code(statusCode);
           }
           // custom redirect https://github.com/dwyl/hapi-error/issues/5
-          if (config.statusCodes[statusCode]
-            && config.statusCodes[statusCode].redirect) {
+          var currentCodeConfig = config.statusCodes[statusCode];
```
Comment (nelsonic) [inline lib/index.js:113]: I _agree_ that this code can be tidied up. I wrote it a _very_ long time ago and it isn't great. 

However, changing quite a lot of code that is not relevant to the bugfix of the **`redirect url`** makes it more work to review the pull request. ⏳ 

This kind of refactoring absolutely has a place in a "refactoring" pull request, but not in a "bugfix" where the purpose is to change the _minimum_ amount of code in order to fix a _specific_ bug.

File: lib/index.js [hunk 2]
```diff
@@ -106,24 +110,25 @@ exports.plugin = {
...
+            if (isFunction(currentCodeConfig.redirect)) {
+              let redirectPath = currentCodeConfig.redirect(request);
               if (redirectPath) {
                 return reply.redirect(redirectPath);
               }
-            } else if (config.statusCodes[statusCode].redirect) { // if parameter is string, append redirect query
-              return reply.redirect(config.statusCodes[statusCode].redirect + '?redirect=' + request.url.path);
+            } else if (currentCodeConfig.redirect) { // if parameter is string, append redirect query
+              var redirectString = request.url.pathname + request.url.search;
+              return reply.redirect(currentCodeConfig.redirect + '?redirect=' + redirectString);
```
Comment (nelsonic) [inline lib/index.js:123]: This appears to solve the issue of having an `undefined` value in the `redirect` URL param https://github.com/dwyl/hapi-error/issues/69 👍

**Review by nelsonic — APPROVED** [review]
@Hydrock thanks for adding the lines to prevent `undefined` in the `redirect`. 👍

File: lib/index.js (hunk 1 above)
Comment (Hydrock) [reply lib/index.js:113]: ok, i understand you

**Pull request merged**